File: php/OOP/Point.php
<?php

require_once __DIR__.'/Config.php';

class Point
{
    public $isAlive;

    public function __construct(bool $isAlive)
    {
        $this->isAlive = $isAlive;
    }

    public function toString(Config $config): string
    {
        return ($this->isAlive) ? $config->get('alive') : $config->get('dead');
    }
}

File: java/src/Point.java
public class Point {
    public boolean isAlive = false;

    public Point(boolean isAlive) {
        this.isAlive = isAlive;
    }

    public String toString() {
        return this.isAlive ? Game.aliveStr : Game.deadStr;
    }

    public int toInt() {
        return this.isAlive ? 1 : 0;
    }
}
